client/hostinfo_unix.cpp

```diff
@@ -485,7 +485,7 @@ static void parse_meminfo_linux(HOST_INFO& host) {
 // See http://people.nl.linux.org/~hch/cpuinfo/ for some examples.
 //
 static void parse_cpuinfo_linux(HOST_INFO& host) {
-    char buf[1024], features[P_FEATURES_SIZE], model_buf[1024];
+    char buf[P_FEATURES_SIZE], features[P_FEATURES_SIZE], model_buf[1024];
```

Comment:
You increased the size of `buf` here, but on the line 530 https://github.com/BOINC/boinc/pull/5123/files#diff-b710fcc072350b8bbb1bc13eb394da5cdbc0b0dfab019a93890ec10eadb41b86R530
we still read a chunk of 1024. And while this is generally won't fail, that could still mean that some data could be truncated and/or corrupted.
Please fix this.

Reply:
Good catch; I changed it to  sizeof(buf)